api/app/signals/apps/signals/v1/serializers.py

```diff
@@ -95,3 +85,255 @@ class Meta:
+    priority = _NestedPriorityModelSerializer(required=False)
+    notes = _NestedNoteModelSerializer(many=True, required=False)
+
+    class Meta:
+        model = Signal
+        fields = (
+            '_links',
+            '_display',
+            'category',
+            'id',
+            'image',
+            'location',
+            'status',
+            'reporter',
+            'priority',
+            'notes',
+        )
+        read_only_fields = (
+            'id',
+            'image',
+        )
+
+    def update(self, validated_data):
+        if 'location' in validated_data and validated_data['location']:
+            location_data = validated_data.pop('location')
+            location_data['created_by'] = self.context['request'].user
+            Signal.actions.update_location()
+
+
+class PrivateSignalSerializerList(HALSerializer):
+    serializer_url_field = PrivateSignalLinksField
+    _display = DisplayField()
+
+    image = serializers.ImageField(read_only=True)
+
+    location = _NestedLocationModelSerializer()
+    status = _NestedStatusModelSerializer(required=False)
+    category = _NestedCategoryModelSerializer(source='category_assignment')
+    reporter = _NestedReporterModelSerializer()
+    priority = _NestedPriorityModelSerializer(required=False)
+    notes = _NestedNoteModelSerializer(many=True, required=False)
+
+    class Meta:
+        model = Signal
+        fields = (
+            '_links',
+            '_display',
+            'id',
+            'signal_id',
+            'source',
+            'text',
+            'text_extra',
+            'status',
+            'location',
+            'category',
+            'reporter',
+            'priority',
+            'created_at',
+            'updated_at',
+            'incident_date_start',
+            'incident_date_end',
+            'operational_date',
+            'image',
+            'extra_properties',
+            'notes',
+        )
+        read_only_fields = (
+            'created_at',
+            'updated_at',
+        )
+
+    def validate(self, data):
+        # for debugging
+        # print('\nData received\n', data, '\n')
+        return super().validate(data)
+
+    def create(self, validated_data):
+        # We ignore the status from the incoming Signal and replace it with the
+        # initial state in the workflow (GEMELD).
+        logged_in_user = self.context['request'].user
+        INITIAL_STATUS = {
+            'state': workflow.GEMELD,  # see models.py is already default
+            'text': None,
+            'user': logged_in_user.email,
+        }
+        validated_data.pop('status', None)  # Discard what was sent over the API
+
+        # We require location and reporter to be set and to be valid.
+        reporter_data = validated_data.pop('reporter')
+
+        location_data = validated_data.pop('location')
+        location_data['created_by'] = logged_in_user.email
```

Comment:
Could also be solved using something like this on the serializer

`created_by = serializers.PrimaryKeyRelatedField(
    read_only=True, 
    default=serializers.CurrentUserDefault()
)`

Reply:
@tcoenen this could be done for all serializers that require a created_by

Reply:
@tcoenen just realised we are only using the email from the loggedin user. So what I stated above this will not work in the current implementation.